labeling/internal/python.go

```diff
@@ -107,22 +109,65 @@ func fileContainsString(c codebase.Codebase, filePath string, str string) bool {
 }
 
 func getPythonVersion(c codebase.Codebase) string {
-	versionRegex := regexp.MustCompile(`[0-9.]+`)
 
 	versionFilePath, _ := c.FindFile(".python-version")
-	if versionFilePath == "" {
-		return ""
-	}
+	if versionFilePath != "" {
+		file, err := c.ReadFile(versionFilePath)
+		if err != nil {
+			log.Println("Unable to read file .python-version. Attempting to determine python version another way... ", err)
+		} else {
+			versionRegex := regexp.MustCompile(`[0-9.]+`)
+			pythonVersion := versionRegex.FindString(string(file))
+			if pythonVersion != "" {
+				log.Println("Unable to parse file .python-version. Attempting to determine python version another way...")
 
-	file, err := c.ReadFile(versionFilePath)
-	if err != nil {
-		return ""
+			}
+			log.Println("Found Python version in .python-version file: ", pythonVersion)
+			return pythonVersion
+		}
 	}
 
-	version := versionRegex.FindString(string(file))
-	if version != "" {
-		return version
+	pyprojectFilePath, _ := c.FindFile("pyproject.toml")
+	if pyprojectFilePath != "" {
+		file, err := c.ReadFile(pyprojectFilePath)
+		if err != nil {
+			log.Println("Unable to read file pyproject.toml. Attempting to determine python version another way... ", err)
+		} else {
+			tree, err := toml.LoadBytes(file)
+			if err != nil {
+				log.Println("Unable to read file pyproject.toml. Attempting to determine python version another way... ", err)
+			}
+			pythonVersion := tree.Get("tool.poetry.dependencies.python")
+			if pythonVersion == nil || pythonVersion.(string) == "" {
+				log.Println("Error parsing pyproject.toml, python version is nil")
+			} else {
+				pythonVersion = strings.TrimPrefix(pythonVersion.(string), "^")
+				log.Println("Found Python version in pyproject.toml file: ", pythonVersion.(string))
+				return pythonVersion.(string)
+			}
+		}
 	}
 
+	pipfileFilePath, _ := c.FindFile("Pipfile")
+	if pipfileFilePath != "" {
+		file, err := c.ReadFile(pipfileFilePath)
+		if err != nil {
+			log.Println("Unable to read file Pipfile. Attempting to determine python version another way... ", err)
+		} else {
+			tree, err := toml.LoadBytes(file)
```

Comment:
Is this attempting to read the same project file as above again?

Reply:
on line 153 you mean? it's reading the file in as []byte and then the LoadByte turns that data into a tree that we parse

Reply:
Got it - I didn't realize pipfile was also toml - thanks.

Reply:
ohhhh I see what you're saying, ya the go-toml is misleading because it can also read the Pipfile